Sources/MongoSwift/Operations/ListCollectionsOperation.swift

```diff
@@ -117,38 +117,36 @@ internal struct ListCollectionsOperation: Operation {
             }
         }
 
-        let indexes: OpaquePointer = self.database.withMongocDatabase(from: connection) { dbPtr in
+        let collections: OpaquePointer = self.database.withMongocDatabase(from: connection) { dbPtr in
             guard let collections = mongoc_database_find_collections_with_opts(dbPtr, opts._bson) else {
                 fatalError(failedToRetrieveCursorMessage)
             }
             return collections
         }
 
         if self.nameOnly {
-            let cursor: MongoCursor<Document> = try MongoCursor(
-                stealing: indexes,
+            let cursor = try Cursor(
```

Comment:
with this new "internal cursor API", we can now operate on the internal cursor type, rather than calling internal methods on our public `MongoCursor` type.